lib/src/scss-shared/_nx-input-helpers.scss

```diff
@@ -34,6 +37,12 @@
       }
     }
 
+    .nx-checkbox__input:checked {
+      + .nx-checkbox__box {
```

Comment:
This really goes against the point of this being a mixin

Reply:
> This really goes against the point of this being a mixin

Similar to the NxRadio button mixin issues the interactions between `:focus`, `:hover`, and `:checked` states are now so complicated that this needs to be here because of specificity.

I like all of the `:focus` styles being collected in one place since (for the most part) they are shared so it is DRY-er. But obviously we have the NxRadio and NxCheckbox specific styles to deal with.

We could get rid of the mixin altogether, and just have some duplicated styles between NXRadio and NxCheckbox.

We could recreate the whole cascade in each component's SCSS to deal with the inheritance issues (I dislike that idea) and it negates the benefits of the mixin.

There is a little bit of refining that could be done after both NxRadio and NxCheckbox are merged to the feature branch (grouping both `:checked` styles into one group for example, but this won't eliminate the specific styles).

I'm open to suggestions.